base/show.jl

```diff
@@ -802,10 +802,18 @@ function show_unquoted(io::IO, ex::Expr, indent::Int, prec::Int)
             if (na == 2 || (na > 2 && func in (:+, :++, :*))) &&
                     all(!isa(a, Expr) || a.head !== :... for a in func_args)
                 sep = " $func "
+                # parenthesize (-x) ^ y for negative x
```

Comment:
It seems weird to single out `-` and `^` here.  Should we parenthesize pretty much any case of `(unary x) OP y`?